### ISSUE TITLE
Compile Nim from Mix

### ISSUE THREAD

**Issue by elcritch** [wltsmrz/nimler#18]
I've started another project using Nimler. Thanks for the library again!

In your examples you don't use Make or anything to tie in Nim compilation with an Elixir project's mix setup. Here's on simple way you could add somewhere in your documentation: 

```elixir
defmodule ExampleNifs.MixProject do
  use Mix.Project

  def project do
    [
      app: :widgetlords_nifs,
      version: "0.1.0",
      elixir: "~> 1.9",
      ...
      aliases: [nim: [&nim_c/1]],
      deps: deps()
    ]
  end

  ...

  defp nim_c([]) do
    case System.find_executable("nim") do
      nil ->
        Mix.Shell.IO.info("Install Nim to generate C code.")

      nimc ->
        System.cmd(nimc, ["c", "--compile_only", ...], cd: "nimsrc/", into: IO.stream(:stdio, :line))
    end
  end
end
```

Ultimately it'd be nice to have a mix task that could be installed globally, but this works pretty well. You could also modify the "compile"  alias to call the Nim compile step and then the regular Elixir compilation step. 

This can be done by something like:
```elixir
      aliases: [nim: [&nim_c/1], compile: ["nim", "compile"]],
```

**Comment by wltsmrz** [comment]
> I've started another project using Nimler. Thanks for the library again!

You're welcome. I started on nimler when I couldn't get any public examples of Rustler working with the (at the time) latest Elixir/OTP, and I'm not adept enough (yet?) with Rust to fix Rust things. I figure nim is a reasonably good fit for writing NIFs--not perfect but pretty good.

> In your examples you don't use Make or anything to tie in Nim compilation with an Elixir project's mix setup

Agreed. I have a todo list somewhere (the project isn't abandoned or anything). Making things easier from the Elixir side is on the list. I'll experiment with your recommendation. I also reserved [nimler](https://hex.pm/packages/nimler) a while back in hex if we need any fixin's.
